### PR TITLE
catch probable auth error and enables retry

### PR DESCRIPTION
**Pull request by allensallinger**
catch probable auth error and enables retry
test for probable auth error has been added as well.

### CODE REVIEW

File: great_expectations_cloud/agent/agent.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def _listen(self) -> None:
             print("Received request to shutdown.")
         except (SubscriberError, ClientError):
             print("Connection to GX Cloud has encountered an error.")
-        except AuthenticationError:
+        except (AuthenticationError, ProbableAuthenticationError):
```
Comment (anthonyburdi) [inline great_expectations_cloud/agent/agent.py:147]: I think we need to also add `ProbableAuthenticationError` to the `@retry` decorator above on line 126 to enable the retry: https://tenacity.readthedocs.io/en/stable/api.html#tenacity.retry.retry_if_exception_type

**Review by anthonyburdi — CHANGES_REQUESTED** [review]
Thanks for handling! Looks good just one small edit.

File: great_expectations_cloud/agent/agent.py (hunk 1 above)
Comment (allensallinger) [reply great_expectations_cloud/agent/agent.py:147]: Ahh, fixed that and fixed up the tests. Thanks for the review.

**Review by anthonyburdi — APPROVED** [review]
Thanks for fixing!

**Pull request merged**